### DISCUSSION TITLE
Sharing OneDrive folder with a link or an invite by email failing.

### DISCUSSION THREAD

**Discussion by nwold** [O365/python-o365#650] [Q&A]
### Share with link / invite
Hi am trying to give someone outside my org access to a folder in my OneDrive. I have been successful creating a folder, uploading files, and sending an email. I have not been successful generating a share link or sharing by invite. The following code is throwing the error below:
`permission = requestFolder.share_with_link(share_type='view', share_scope = 'anonymous')` 
and 
`permission = requestFolder.share_with_invite(recipients='my@email.com', require_sign_in=False, send_email=True, message='Greetings!!', share_type='view')`.

I get the message: "Error Message: The feature has been disabled. Please contact your admin to get it enabled."  I am not sure what this means because I am able to share the the folder with an email address manually.
I have the scopes set:
`scopes = ['Mail.ReadWrite', 'Mail.Send', 'offline_access', 'User.Read', 'Files.Read.All', 'Files.ReadWrite.All', 'Tasks.Read', 'Tasks.ReadWrite', 'Sites.Read.All', 'Sites.ReadWrite.All']`

**Comment by alejcas** [comment]
It should work. It was working for me before.

The message seems to indicate that something must be allowed by the admin. Maybe this setting is disabled for the graph api... i don't really know.

**Reply by nwold** [reply]
To share with link, I had to change 'anonymous' to 'organization'.  And to share with invite, I had to require sign in. This allows me to send an invite via email to anyone outside the organization. Amazing what a weekend away can do to help solving a problem. Thanks for this Library. It has been extremely helpful.
